src/base_alloc/base_alloc_linear.c

```diff
@@ -19,24 +19,56 @@
 // alignment of the linear base allocator
 #define MEMORY_ALIGNMENT (sizeof(uintptr_t))
 
-// metadata of the linear base allocator
-typedef struct {
-    size_t pool_size;
+typedef struct umf_ba_next_linear_pool_t umf_ba_next_linear_pool_t;
+
+// metadata is set and used only in the main (the first) pool
+typedef struct umf_ba_main_linear_pool_meta_t {
+    size_t pool_size; // size of each pool (argument of each ba_os_alloc() call)
     os_mutex_t lock;
     char *data_ptr;
     size_t size_left;
+#ifndef NDEBUG
+    size_t n_pools;
+#endif /* NDEBUG */
 } umf_ba_main_linear_pool_meta_t;
 
-// pool of the linear base allocator
+// the main pool of the linear base allocator (there is only one such pool)
 struct umf_ba_linear_pool {
+    // address of the beginning of the next pool (a list of allocated pools to be freed in umf_ba_linear_destroy())
```

Comment:
please break long comments manually - this is disabled in clang-format

Reply:
Done